### DISCUSSION TITLE
API endpoint for updating role permission

### DISCUSSION THREAD

**Discussion by oOhyeahh** [apache/superset#26247] [Ideas]
Could we add an api endpoint to allow admin user to update the role?  I have a user case that assign new permission to my custom role when a new dataset is created. And I can't to just user to access all datasources.c

**Comment by christophatinnuce** [comment]
Have a look at:

FAB_ADD_SECURITY_API

https://flask-appbuilder.readthedocs.io/en/latest/config.html

supersethost/swagger/v1:

![grafik](https://github.com/apache/superset/assets/153641505/617a5159-1071-46f3-bdb9-de3be4fa608a)

**Comment by oOhyeahh** [comment]
Yes. It's looks the right direction. I also notice I have to manually add the permission view in the superset application to get the data. i.e can_post Role, can_get Role . But it's outside superset scope I guess. I will close it.
